lib/coinbase/transaction.rb

```diff
@@ -0,0 +1,126 @@
+      BROADCAST = 'broadcast'
+
+      # The Transaction is complete and has confirmed on the Network.
+      COMPLETE = 'complete'
+
+      # The Transaction has failed for some reason.
+      FAILED = 'failed'
+
+      # The states that are considered terminal on-chain.
+      TERMINAL_STATES = [COMPLETE, FAILED].freeze
+    end
+
+    # Returns a new Transaction object. Do not use this method directly.
+    # @param model [Coinbase::Client::Transaction] The underlying Transaction object
+    def initialize(model)
+      raise unless model.is_a?(Coinbase::Client::Transaction)
+
+      @model = model
+    end
+
+    # Returns the Unsigned Payload of the Transaction.
+    # @return [String] The Unsigned Payload
+    def unsigned_payload
+      @model.unsigned_payload
+    end
+
+    # Returns the Signed Payload of the Transaction.
+    # @return [String] The Signed Payload
+    def signed_payload
+      @model.signed_payload
+    end
+
+    # Returns the Transaction Hash of the Transaction.
+    # @return [String] The Transaction Hash
+    def transaction_hash
+      @model.transaction_hash
+    end
+
+    # Returns the status of the Transaction.
+    # @return [Symbol] The status
+    def status
+      @model.status
+    end
+
+    # Returns the from address for the Transaction.
+    # @return [String] The from address
+    def from_address_id
+      @model.from_address_id
+    end
+
+    # Returns whether the Transaction is in a terminal state.
+    # @return [Boolean] Whether the Transaction is in a terminal state
+    def terminal_state?
+      Status::TERMINAL_STATES.include?(status)
+    end
+
+    # Returns the link to the transaction on the blockchain explorer.
+    # @return [String] The link to the transaction on the blockchain explorer
+    def transaction_link
+      # TODO: Parameterize this by Network.
+      "https://sepolia.basescan.org/tx/#{transaction_hash}"
+    end
+
+    # Returns the underlying raw transaction.
+    # @return [Eth::Tx::Eip1559] The raw transaction
+    def raw
+      return @raw unless @raw.nil?
+
+      raw_payload = [unsigned_payload].pack('H*')
+      parsed_payload = JSON.parse(raw_payload)
+
+      params = {
+        chain_id: parsed_payload['chainId'].to_i(16),
+        nonce: parsed_payload['nonce'].to_i(16),
+        priority_fee: parsed_payload['maxPriorityFeePerGas'].to_i(16),
+        max_gas_fee: parsed_payload['maxFeePerGas'].to_i(16),
+        gas_limit: parsed_payload['gas'].to_i(16), # TODO: Handle multiple currencies.
+        from: Eth::Address.new(from_address_id),
+        to: Eth::Address.new(parsed_payload['to']),
+        value: parsed_payload['value'].to_i(16),
+        data: parsed_payload['input'] || ''
+      }
+
+      @raw = Eth::Tx::Eip1559.new(Eth::Tx.validate_eip1559_params(params))
+    end
+
+    # Signs the Transaction with the provided key and returns the hex signing payload.
+    # @return [String] The hex-encoded signed payload
+    def sign(key)
+      raw.sign(key)
```

Comment:
@alex-stone should we update `@model.signed_payload` here?

Reply:
Comment left about this here: https://github.com/coinbase/coinbase-sdk-ruby/pull/73

I think that it makes sense to do that at some point given in the end-user custodial paradigm we will be broadcasting back the full transaction effectively.

I think we can iterate on that as we don't actually source the signed payload of this currently.

But yeah that's definitely an option!